File: langs/typescript/src/lib/hello-vmx.ts
import {
  ComponentVMOf,
  NullMessageHub,
  NullDispatcher,
} from 'vmx';

/**
 * Smoke-test VMx wiring by instantiating a trivial ComponentVMOf and
 * returning a descriptive string for the UI.
 *
 * VMx uses a builder pattern — ComponentVM has no public constructor.
 * NullMessageHub and NullDispatcher are the null-object singletons
 * provided by VMx for tests and standalone demos.
 */
export function helloVMx(): string {
  const vm = ComponentVMOf.builder<{ message: string }>()
    .name('hello')
    .model({ message: 'hello from VMx' })
    .services(NullMessageHub.INSTANCE, NullDispatcher.INSTANCE)
    .modeledHinter((m) => m.message)
    .build();

  vm.construct();

  return `VMx loaded — model.message = "${vm.model.message}", status = ${vm.status}`;
}

File: langs/csharp/src/GuideArch.Models/HelloVmx.cs
using VMx.Components;
using VMx.Services;

namespace GuideArch.Models;

/// <summary>
/// Smoke-test VMx wiring by instantiating a trivial ComponentVM and
/// returning a descriptive string for the UI.
/// </summary>
public static class HelloVmx
{
    public sealed record Greeting(string Message);

    public static string Run()
    {
        var vm = ComponentVM<Greeting>.Builder()
            .Name("hello")
            .Model(new Greeting("hello from VMx"))
            .Services(NullMessageHub.Instance, NullDispatcher.Instance)
            .Build();

        vm.Construct();
        return $"VMx loaded — model.Message = \"{vm.Model.Message}\", status = {vm.Status}";
    }
}
